packages/cypress/src/steps.ts

```diff
@@ -96,12 +97,20 @@ function getTestsFromResults(
         afterEach: [],
         main: [],
       },
-      error: result.displayError
-        ? {
-            name: "DisplayError",
-            message: result.displayError,
-          }
-        : null,
+      // attempt.error is available here:
+      // https://github.com/cypress-io/cypress/blob/61156808413be8b99264026323ce3abfb22c4c26/packages/server/lib/modes/results.ts#L20
+      // but it's lost when creating a public test result:
+      // https://github.com/cypress-io/cypress/blob/61156808413be8b99264026323ce3abfb22c4c26/packages/server/lib/modes/results.ts#L111
+      // `.displayError` represents the error reported by the last attempt
+      // for all of the previous attempts we rely on the search for the last errored step in `groupStepsByTest`
+      // if an error is found there, it will be set on the test, the information set here is not overriden though
```

Comment:
This logic is a little bit wonky but it makes sense. The `.displayError` is something computed by Cypress so it should have higher accuracy than what we can compute on our own. OTOH, we have to compute this on our own for non-last attempts because Cypress doesn't expose this information to us.

It would be great to refactor this so we don't have to rely on a mutation *elsewhere* but that's out of this PR's scope